mewarpx/mewarpx/mcc_wrapper.py

```
@@ -0,0 +1,90 @@
+                            "ion_scattering.dat": "elastic",
+                            "ion_back_scatter.dat": "back",
+                            "charge_exchange.dat": "charge_exchange"
+                            }
+        requires_energy = {
+                            "Ar": {
+                                "excitation_1.dat": 11.5,
+                                "ionization.dat": 15.7596112
+                                },
+                            "He": {
+                                "excitation_1.dat": 19.82,
+                                "excitation_2.dat": 20.61,
+                                "ionization.dat": 24.55
+                                },
+                            "Xe": {
+                                "excitation_1.dat": 8.315,
+                                "ionization.dat": 12.1298431
+                                }
+                        }
+
+        # build scattering process dictionaries
+        elec_scattering_processes = {}
+        ion_scattering_processes = {}
+
+        for path in range(len(file_paths)):
+            file_name = os.path.absename(path)
+            # if electron process
+            if file_name in elec_collision_types:
+                scatter_dict = {"cross_section": path}
+                # add energy if needed
+                if file_name in requires_energy[self.ion_species.name]:
+                    scatter_dict["energy"] = requires_energy[self.ion_species.name][file_name]
+                # specify species for ionization
+                if file_name == "ionization.dat":
+                    scatter_dict["species"] = self.ion_species
+                elec_scattering_processes[elec_collision_types[file_name]] = scatter_dict
+            # if ion process
+            elif file_name in ion_collision_types:
+                scatter_dict = {"cross_section": path}
+                ion_scattering_processes[ion_collision_types[file_name]] = scatter_dict
```

else raise an error the filename wasn't recognized? Otherwise filename typos will silently drop that type of interaction.

---

The best way I could think of implementing this is putting an else statement, but this means that the directory can't have any unexpected .dat files in it, or it will throw an error.

---

Yeah, I think that's still the best alternative; other *.dat files should just be in other directories in all other cases.